File: problems/2772_apply-operations-to-make-all-array-elements-equal-to-zero/solution.java
class Solution {

    public boolean checkArray(int[] nums, int k) {
        int n = nums.length;
        long[] diff = new long[n + 1];
        long running = 0;
        for (int i = 0; i < n; i++) {
            running += diff[i];
            long cur = nums[i] - running;
            if (cur < 0) {
                return false;
            }
            if (cur == 0) {
                continue;
            }
            if (i + k > n) {
                return false;
            }
            running += cur;
            diff[i + k] -= cur;
        }
        return true;
    }
}

File: problems/2772_apply-operations-to-make-all-array-elements-equal-to-zero/solution.py
from typing import List, Optional


class Solution:
    def checkArray(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        diff = [0] * (n + 1)
        running = 0
        for i in range(n):
            running += diff[i]
            cur = nums[i] - running
            if cur < 0:
                return False
            if cur == 0:
                continue
            if i + k > n:
                return False
            running += cur
            diff[i + k] -= cur
        return True
